src/JMS/Serializer/Metadata/Driver/YamlDriver.php

```diff
@@ -292,7 +292,9 @@ private function addClassProperties(ClassMetadata $metadata, array $config)
                     throw new RuntimeException('The "map" attribute must be set, and be an array for discriminators.');
                 }
                 $groups = isset($config['discriminator']['groups']) ? $config['discriminator']['groups'] : array();
-                $metadata->setDiscriminator($config['discriminator']['field_name'], $config['discriminator']['map'], $groups);
+                $strict_deserialize = isset($config['discriminator']['strict_deserialize']) ? $config['discriminator']['strict_deserialize'] : true;
```

Comment:
`$config['discriminator']['strict_deserialize']` is a string, make it a boolean here